docs/proposals/2841-flux-hpc/README.md

```diff
@@ -0,0 +1,185 @@
+    // +kubebuilder:default=1
+    // +optional
+    NumProcPerNode *int32 `json:"numProcPerNode,omitempty"`
+
+    // Container image to use for Flux view that installs Flux
+    // This must be compatible with the application container
+    // Get the flux view container (these are choices)
+    // ghcr.io/converged-computing/flux-view-rocky:arm-9
+    // ghcr.io/converged-computing/flux-view-rocky:arn-8
+    // ghcr.io/converged-computing/flux-view-rocky:tag-9
+    // ghcr.io/converged-computing/flux-view-rocky:tag-8
+    // ghcr.io/converged-computing/flux-view-ubuntu:tag-noble
+    // ghcr.io/converged-computing/flux-view-ubuntu:tag-jammy
+    // ghcr.io/converged-computing/flux-view-ubuntu:tag-focal
+    // ghcr.io/converged-computing/flux-view-ubuntu:arm-jammy
+    // ghcr.io/converged-computing/flux-view-ubuntu:arm-focal
+    // We use an ubuntu (more recent) default since it is common
+    // +kubebuilder:default="ghcr.io/converged-computing/flux-view-ubuntu:arm-jammy"
+    Image string `json:"image,omitempty"`
+
+    // Network device for flux to use
+    // +kubebuilder:default="eth0"
+    NetworkDevice string `json:"networkDevice,omitempty"`
+
+    // Queue policy for Flux to use
+    // +kubebuilder:default="fcfs"
+    QueuePolicy string `json:"queuePolicy,omitempty"`
+}
+```
+
+Note that we are not exposing interactive or other options that are widely used, but not required.
+
+**Example `ClusterTrainingRuntime`:**
+
+This is an example created by an administrator.
+
+```yaml
+apiVersion: trainer.kubeflow.org/v1alpha1
+kind: ClusterTrainingRuntime
+metadata:
+  name: flux-runtime
+  labels:
+    trainer.kubeflow.org/framework: flux
+spec:
+  mlPolicy:
+    flux:
+      numNodes: 1
+  template:
+    spec:
+      replicatedJobs:
+        - name: trainer
+          template:
+            spec:
+              completionMode: Indexed
+              completions: 1
+              parallelism: 1
+              template:
+                spec:
+                  containers:
+                    - name: node
+                      image: "placeholder-image"
+```
+
+**Example Consuming `TrainJob` YAML:**
+
+```yaml
+apiVersion: trainer.kubeflow.org/v1alpha1
+kind: TrainJob
+metadata:
+  name: lammps-flux-interactive
+spec:
+  # Reference the pre-defined runtime by name
+  runtimeRef:
+    apiGroup: trainer.kubeflow.org
+    name: flux-runtime
+    kind: ClusterTrainingRuntime
+  trainer:
+    numNodes: 4
+    image: ghcr.io/converged-computing/metric-lammps:latest
+```
+
+### Implementation Details
+
+1.  **Controller Logic:** When reconciling a `TrainJob`, the controller fetches the referenced `TrainingRuntime` or `ClusterTrainingRuntime`. It then passes the entire `spec` of the runtime, including the `mlPolicy`, to the plugin framework via the `runtime.Info` struct.
+
+2.  **Plugin Logic:** We add components of Flux via `Build` if the user has defined a Flux policy. The design integrates with the existing Kubeflow Trainer API.
```

Comment:
Do we need to create any additional resources to bootstrap Flux? For example, ConfigMap, Secrets, PVCs, etc.?

Reply:
Yes - a config map and the shared empty directory. I'll add that.